csharp/authenticate/authenticate/Program.cs

```diff
@@ -32,17 +34,26 @@ static void Main(string[] args)
             {
                 if(database.Users.ContainsKey(username)) ExitWithError("User Exists");
 
-                //TODO: the following line adds a new user to the database
-                //      you may with to change how this works for your program.
-                database.Users[username] = new User() { UserName = username };
+                var salt = BCrypt.Net.BCrypt.GenerateSalt(4);
```

Comment:
Is 4 good? Is 10 better? Why is this only performed on the salt and not the hashing itself? 

I don't obviously understand the api of the library I chose.